src/components/Modal/ConfirmationModal.tsx

```diff
@@ -0,0 +1,129 @@
+'use client'
+
+import { ReactNode, useRef } from 'react'
+import { motion, AnimatePresence } from 'motion/react'
+import { createPortal } from 'react-dom'
+import modalBg from './images/modal-bg.svg'
+import { cn } from '@/lib/utils'
+import { HeaderTitle } from '../Typography'
+import { Button } from '../Button'
+import { useClickOutside } from '@/shared/hooks/useClickOutside'
+
+// transition animation duration
+const duration = 0.3
+
+interface ConfirmationModalProps {
+  /**
+   * Modal title displayed at the top
+   */
+  title?: string
+  /**
+   * Pass the main modal text to the `children` prop
+   */
+  children?: ReactNode
+  /**
+   * Called when `I Agree` button is clicked
+   */
+  onAccept?: () => void
+  /**
+   * Called when `I Disagree` button is clicked
+   */
+  onDecline?: () => void
+  /**
+   * When defined, replaces accept/decline buttons
+   */
+  buttons?: ReactNode
+  /**
+   * Controls whether the modal is visible
+   */
+  isOpen: boolean
+  /**
+   * Callback triggered when the modal is closed
+   */
+  onClose: () => void
+  /**
+   * Test identifier for unit and integration tests
+   */
+  'data-testid'?: string
+}
+
+/**
+ * Generic reusable animated modal component.
+ * Can be used for disclaimers, warnings, confirmations, or other modal dialogs.
```

Comment:
Maybe changing this description as the name of the component was changed?

Reply:
changed